docs: add GSoC 2025 project ideas (#1706)
asyncapi/community: mentorship/summerofcode/2025/asyncapi-gsoc-ideas-page.md

> # AsyncAPI Ideas Page: Google Summer of Code 2024

Reviewer: is this not a proposal for 2025 @AceTheCreator 👁️

> # AsyncAPI Ideas Page: Google Summer of Code 2024

Author: Thanks, @thulieblack, i keep forgetting we are in a new year 😄

> Welcome to the **AsyncAPI Ideas Page** with our proposed projects for Google Summer of Code (GSoC) 2024! If you are an interested student/contributor, please don't hesitate to contact our mentors directly to discuss project ideas.

Reviewer: ```suggestion
Welcome to the **AsyncAPI Ideas Page** with our proposed projects for Google Summer of Code (GSoC) 2025! If you are an interested contributor, please don't hesitate to contact our mentors directly to discuss project ideas.
```

- 🧩 **Difficulty:** Medium/Hard
- 👩🏿‍🏫 **Mentor(s):** [@AayushSaini101](https://github.com/AayushSaini101) | [@Souvikns](https://github.com/Souvikns)
- ⏳ **Length:** 175 Hours

## 2) [AI-Powered Assistant for AsyncAPI](https://github.com/asyncapi/chatbot/issues/109)
Build an AI-powered assistant fine-tuned on AsyncAPI to provide accurate answers, generate code snippets, debug specifications, and recommend best practices.

- 🎯 **Outcome:** A fine-tuned LLM-powered chatbot integrated with AsyncAPI’s ecosystem for enhanced developer support.
- 🛠️ **Skills Required:** Javascript/Typescript, Machine Learning (LLMs), NLP, OpenAI/Llama, Chatbot Integration.
- 🧩 **Difficulty:** Medium/Hard
- 👩🏿‍🏫 **Mentor(s):** [@AceTheCreator](https://github.com/AceTheCreator)
- ⏳ **Length:** 175 Hours

## 3) [AsyncAPI Generator Maintainership](https://github.com/asyncapi/generator/issues/1360)
This initiative aims to guide you on a journey from being a contributor to becoming a maintainer of the project. You'll gain insight into the responsibilities of a maintainer, involving tasks beyond mere coding.

> - 🎯 **Outcome:** Taking responsibility for the project's future and continuous improvement.

Reviewer: ```suggestion
- 🎯 **Outcome:** Responsible for the project's future and continuous improvement.
```

- 🛠️ **Skills:** JavaScript/TypeScript, testing libraries, Docker, virtualization, and test automation.
- 🧩 **Difficulty:** Medium/Hard
- 👩🏿‍🏫 **Mentor(s):** [@derberg](https://github.com/derberg)
- ⏳ **Length:** 350 Hours

## 4) [AsyncAPI Conference Website UI Kit Development](https://github.com/asyncapi/conference-website/issues/551)
Develop a comprehensive UI Kit to enhance design consistency, modularity, and maintainability of the AsyncAPI Conference website.

- 🎯 **Outcome:** A structured UI Kit with reusable components, Storybook integration, and improved design consistency.
- 🛠️ **Skills Required:** React, TypeScript, Storybook, UI/UX Design, Component Development.
- 🧩 **Difficulty:** Medium
- 👩🏿‍🏫 **Mentor(s):** [@AceTheCreator](https://github.com/AceTheCreator)
- ⏳ **Length:** 175 Hours

## 5) [VS Code Extension Maintainership](https://github.com/asyncapi/conference-website/issues/551)

> This initiative will guide you from being a contributor to becoming a maintainer of the [VS Code AsyncAPI Preview extension](https://github.com/asyncapi/vs-asyncapi-preview). You'll learn the responsibilities of a maintainer, including code contributions, issue triaging, release management, and community engagement.

Reviewer: ```suggestion
This initiative will guide you from contributing to becoming a maintainer of the [VS Code AsyncAPI Preview extension](https://github.com/asyncapi/vs-asyncapi-preview). You'll learn the responsibilities of a maintainer, including code contributions, issue triaging, release management, and community engagement.
```

- 🧩 **Difficulty:** Medium/Hard
- 👩🏿‍🏫 **Mentor(s):** [@ivangsa](https://github.com/ivangsa)
- ⏳ **Length:** 350 Hours

## 6) Java + Quarkus Template for AsyncAPI Generator
Develop a new AsyncAPI Generator template for Java with Quarkus, leveraging its growing adoption in cloud-native development.

- 🎯 **Outcome:** A fully functional Java + Quarkus template for generating AsyncAPI-based applications.
- 🛠️ **Skills Required:** Java, Quarkus, Templating Engines (Nunjucks/Handlebars), AsyncAPI Generator.
- 🧩 **Difficulty:** Medium/Hard
- 👩🏿‍🏫 **Mentor(s):** [@AayushSaini101](https://github.com/AayushSaini101), [@Souvikns](https://github.com/Souvikns)
- ⏳ **Length:** 350 Hours

## Contact AsyncAPI Mentors
- Join [our Slack workspace](https://www.asyncapi.com/slack-invite).  Observe our [Slack etiquette](https://github.com/asyncapi/.github/blob/master/slack-etiquette.md) and [AsyncAPI code of conduct](https://github.com/asyncapi/.github/blob/master/CODE_OF_CONDUCT.md).
- Join the dedicated Mentorship channel `#09_mentorships` to meet all other GSoC mentees and mentors.

## FAQ

> 1. **How active are previous GSoC contributors in AsyncAPI?** AsyncAPI participated as a standalone organization in GSoC for the first time in 2024, after previously collaborating with Postman. Many contributors from that cohort remain actively involved in the community, including:

Reviewer: ```suggestion
1. **How active are previous GSoC contributors in AsyncAPI?** AsyncAPI participated as a standalone organization in GSoC for the first time in 2024 after previously collaborating with Postman. Many contributors from that cohort remain actively involved in the community, including:
```

   - **[@ashmit-coder](https://github.com/ashmit-coder)**: Now a maintainer of the [conference-website](https://github.com/asyncapi/conference-website) repository.
   - **[@helios2003](https://github.com/helios2003)**: Actively contributing to the [CLI](https://github.com/asyncapi/cli) and [Studio](https://github.com/asyncapi/studio) repositories.
   - **[@vishvamsinh28](https://github.com/vishvamsinh28)**: Active contributor and code reviewer in the [Website](https://github.com/asyncapi/website) and [conference-website](https://github.com/asyncapi/conference-website) repositories.

   You can learn more about our GSoC 2024 participation in our official [blog post](https://www.asyncapi.com/blog/2024-gsoc-wrap).

2. **Is using ChatGPT for GSoC project proposals allowed?**

>    We advise against using ChatGPT or similar AI tools for your GSoC project proposals with AsyncAPI. Should you choose to use such tools, we require that this be fully disclosed in your application.

Reviewer: ```suggestion
  We advise against using ChatGPT or similar AI tools for your GSoC project proposals with AsyncAPI. If you choose to use such tools, we require that you fully disclose this in your application.
```
